yb-voyager/src/queryparser/traversal_plpgsql.go

```diff
@@ -0,0 +1,230 @@
+		return []string{}, fmt.Errorf("error parsing the json string of stmt-%s: %v", query, err)
+	}
+
+	if len(parsedJsonMapList) == 0 {
+		return []string{}, nil
+	}
+
+	parsedJsonMap := parsedJsonMapList[0]
+
+	function := parsedJsonMap[PLPGSQL_FUNCTION]
+	parsedFunctionMap, ok := function.(map[string]interface{})
+	if !ok {
+		return []string{}, fmt.Errorf("error getting the PlPgSQL_Function field in parsed json-%s", parsedJson)
+	}
+
+	actions := parsedFunctionMap[ACTION]
+	var plPgSqlStatements []string
+	TraversePlPgSQLJson(actions, &plPgSqlStatements)
+	return plPgSqlStatements, nil
+}
+
+/*
+Query example-
+
+	CREATE OR REPLACE FUNCTION func_example_advi(
+	   sender_id INT,
+	   receiver_id INT,
+	   transfer_amount NUMERIC
+
+)
+RETURNS VOID
+LANGUAGE plpgsql
+AS $$
+BEGIN
+
+	-- Acquire advisory locks to prevent concurrent updates on the same accounts
+	PERFORM pg_advisory_lock(sender_id);
+	PERFORM pg_advisory_lock(receiver_id);
+
+	-- Deduct the amount from the sender's account
+	UPDATE accounts
+	SET balance = balance - transfer_amount
+	WHERE account_id = sender_id;
+
+	-- Add the amount to the receiver's account
+	UPDATE accounts
+	SET balance = balance + transfer_amount
+	WHERE account_id = receiver_id;
+
+	-- Release the advisory locks
+	PERFORM pg_advisory_unlock(sender_id);
+	PERFORM pg_advisory_unlock(receiver_id);
+
+END;
+$$;
+
+parsed json -
+
+	{
+	  "PLpgSQL_function": {
+	    "datums": [
+	      ...
+	    ],
+	    "action": {
+	      "PLpgSQL_stmt_block": {
+	        "lineno": 2,
+	        "body": [
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 4,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_lock(sender_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				... similar to above
+	          },
+	          {
+	            "PLpgSQL_stmt_execsql": {
+	              "lineno": 8,
+	              "sqlstmt": {
+	                "PLpgSQL_expr": {
+	                  "query": "UPDATE accounts \n    SET balance = balance - transfer_amount \n    WHERE account_id = sender_id",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				.... similar to above
+	          },
+	          {
+	            ... similar to below
+	          },
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 19,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_unlock(receiver_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+	            "PLpgSQL_stmt_return": {}
+	          }
+	        ]
+	      }
+	    }
+	  }
+	}
+*/
+func TraversePlPgSQLJson(fieldValue interface{}, plPgSqlStatements *[]string) {
+	fieldMap, isMap := fieldValue.(map[string]interface{})
```

Comment:
for now the traverse function seems straightforward. If gets to handle a lot of cases and would need generalisation, consider exploring the implementation using `reflect` pacakge.